lib/envars.ml

```diff
@@ -171,11 +171,15 @@ let xdg_dirs ~warn =
 
 (* Print the configuration information *)
 
+let ensure_trailing_slash s =
+  let n = String.length s in
+  if n <> 0 && s.[n-1] <> '/' then s ^ "/" else s
+
 let print_config ?(prefix_var_name="") f coq_src_subdirs =
   let open Printf in
   fprintf f "%sLOCAL=%s\n" prefix_var_name (if Coq_config.local then "1" else "0");
-  fprintf f "%sCOQLIB=%s/\n" prefix_var_name (coqlib ());
-  fprintf f "%sDOCDIR=%s/\n" prefix_var_name (docdir ());
+  fprintf f "%sCOQLIB=%s\n" prefix_var_name (ensure_trailing_slash (coqlib ()));
+  fprintf f "%sDOCDIR=%s\n" prefix_var_name (ensure_trailing_slash (docdir ()));
```

Comment:
This change seems unrelated, right? Also, is it correct? IIUC these variables are not supposed to have a trailing slash.

Reply:
This is unrelated to the title of the PR. It is related to what `coqc -config` reports (formerly, I observed things like `/mydir//`). It is maybe too minor for being worth a different PR.

Reply:
Umm, this doesn't affect what `coqc -config` outputs, but only what gets written for coq_makefile, right?

The code also doesn't seem correct, I dunno what problem is trying to solve but that should be done then at the level of `coqlib ()`, not here in the printing. I'd suggest we remove this change for now, as this PR has to be backported.

Reply:
> this doesn't affect what coqc -config outputs, but only what gets written for coq_makefile, right?

It is called in `toplevel/coqtop.ml`. I believe it was used in former versions of `coq_makefile`.

I don't know what are the exact specifications of `coqlib ()` and `docdir ()`, i.e. I don't know if they are intended to return a path ended by `/` or not.

> I'd suggest we remove this change for now, as this PR has to be backported.

OK. I'm removing it.

Reply:
> I don't know what are the exact specifications of `coqlib ()` and `docdir ()`, i.e. I don't know if they are intended to return a path ended by `/` or not.

This changed recently; indeed it is always confusing. Sometime ago I tried to follow the approach used in Dune of having an opaque `Path.t` type, but indeed that's pretty challenging to do, on the other hand it brings consistency.